src/Symfony/Component/Routing/Generator/UrlGenerator.php

```diff
@@ -67,7 +67,7 @@ class UrlGenerator implements UrlGeneratorInterface, ConfigurableRequirementsInt
         '%7C' => '|',
     ];
 
-    public function __construct(RouteCollection $routes, RequestContext $context, LoggerInterface $logger = null, string $defaultLocale = null)
+    public function __construct(RouteCollection $routes, RequestContext $context, LoggerInterface $logger = null, string $defaultLocale = 'en')
```

Comment:
This is a BC break + it does not really fix the problem if the route does not have an "en" variation.